libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/RouteArrowViewModelFactory.kt

```diff
@@ -0,0 +1,24 @@
+package com.mapbox.navigation.dropin
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.mapbox.navigation.dropin.viewmodel.RouteArrowViewModel
+import com.mapbox.navigation.ui.maps.route.arrow.api.MapboxRouteArrowApi
+import com.mapbox.navigation.ui.maps.route.arrow.api.MapboxRouteArrowView
+import com.mapbox.navigation.ui.maps.route.arrow.model.RouteArrowOptions
+
+class RouteArrowViewModelFactory(
+    private val routeArrowOptions: RouteArrowOptions
+) : ViewModelProvider.Factory {
+
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(RouteArrowViewModel::class.java)) {
+            @Suppress("UNCHECKED_CAST")
+            return RouteArrowViewModel(
+                MapboxRouteArrowApi(),
+                MapboxRouteArrowView(routeArrowOptions)
```

Comment:
I don't think it's a good idea to pass the view into the view model as it breaks the pattern. ViewModel should not know anything about a view even though in this case it is a custom view and not something that is a part of a layout. Also, if not managed properly it can lead to leaks since view model holds reference to view which usually gets destroyed on configuration change. I would encourage for `MapboxRouteArrowView` to be instantiated in `NavigationView` and it should observe changes to `RouteArrowViewModel`. 
Same applies to `RouteLineViewModel`

Reply:
The MapboxRouteArrowView isn't an android view and doesn't reference any android view.  The RouteLineViewModel and RouteArrowViewModel are delegates for the interaction between the two API classes.  There's no value add for the view model outputting the API response.  

The MapboxRouteArrowView and MapboxRouteLineView classes don't hold any external references so there's nothing to leak.

Reply:
Okay sounds good. We would just have to be sure that going forward we don't add any context or Android View related constructs to `MapboxRouteArrowView` and `MapboxRouteLineView` or else the DropIn module will have chances of leaking.